public/kibana-integrations/kibana-discover.js

```diff
@@ -528,6 +529,8 @@ function discoverController(
           {
             next: () => {
               $scope.filters = filterManager.filters;
+              const customFilterAllowedAgents = getFilterWithAuthorizedAgents(store.getState().appStateReducers.allowedAgents);
```

Comment:
If the `store state` is not ready or if for some reason `getFilterWithAuthorizedAgents` get `undefined` as a parameter, it **returns undefined** and then **it is added as a filter**. This causes several errors which can be avoided by checking if the filter format is valid before adding it.

![image](https://user-images.githubusercontent.com/9343732/113746055-48021680-96dc-11eb-9a5f-424d18e4e709.png)
![image](https://user-images.githubusercontent.com/9343732/113746429-b34be880-96dc-11eb-864c-54d0ec26f68a.png)

Reply:
Perfect, thank you very much for reporting it, it has already been fixed!